account_invoice_ubl/models/account_invoice.py

```diff
@@ -94,7 +94,8 @@ def _ubl_add_legal_monetary_total(self, parent_node, ns, version='2.1'):
         monetary_total = etree.SubElement(
             parent_node, ns['cac'] + 'LegalMonetaryTotal')
         cur_name = self.currency_id.name
-        prec = self.currency_id.decimal_places
+        # prec = self.currency_id.decimal_places
```

Comment:
I would prefer to keep the previous lines not commented, something like this:

```suggestion
        prec = self.currency_id.decimal_places
        prec = prec if prec <= 2 else 2
```

And the same logic for the other changes you made.